CommandState[] active_cmds
CommandState[] finished_cmds